--- CMakeLists.txt      2022-11-09 06:37:21.000000000 -0600
+++ CMakeLists.txt      2024-06-26 09:35:55.336166188 -0500
@@ -187,13 +187,13 @@
     include_directories(${OpenCL_INCLUDE_DIRS})
     message(STATUS "OpenCL include directory: " ${OpenCL_INCLUDE_DIRS})
     if(WIN32)
-        set(Boost_USE_STATIC_LIBS ON)
+        set(Boost_USE_STATIC_LIBS OFF)
     endif()
     find_package(Boost 1.56.0 COMPONENTS filesystem system REQUIRED)
     if(WIN32)
         # disable autolinking in boost
         add_definitions(-DBOOST_ALL_NO_LIB)
     endif()
     include_directories(${Boost_INCLUDE_DIRS})
     add_definitions(-DUSE_GPU)
 endif()
