--- ./CMakeLists.txt    2025-08-24 22:55:26
+++ CMakeLists.txt     2025-08-27 22:03:49
@@ -38,7 +38,7 @@
 if(BUILD_CPP_TEST)
   set(CMAKE_CXX_STANDARD 14)
 else()
-  set(CMAKE_CXX_STANDARD 11)
+  set(CMAKE_CXX_STANDARD 17)
 endif()
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
@@ -255,10 +255,8 @@
 
     add_definitions(-DUSE_CUDA)
 
-    if(NOT DEFINED CMAKE_CUDA_STANDARD)
-      set(CMAKE_CUDA_STANDARD 11)
-      set(CMAKE_CUDA_STANDARD_REQUIRED ON)
-    endif()
+    set(CMAKE_CUDA_STANDARD 17)
+    set(CMAKE_CUDA_STANDARD_REQUIRED ON)
 endif()
 
 include(CheckCXXSourceCompiles)
